geniza/entities/models.py

```diff
@@ -68,6 +70,78 @@ def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
 
 
+class Event(models.Model):
+    """A historical event involving one or more documents, and optionally additional entities."""
+
+    name = models.CharField(
+        max_length=255,
+        help_text="A short name for the event to use as a display label",
+    )
+    description = models.TextField(
+        blank=True,
+        help_text="A description of the event",
+    )
+    footnotes = GenericRelation(Footnote, blank=True, related_query_name="event")
+    display_date = models.CharField(
+        "Display date",
+        help_text='''Display date for manually entered or automatically generated date,
+        as it should appear in the public site, such as "Late 12th century"''',
+        max_length=255,
+        blank=True,  # use standard date for display if this is blank
+    )
+    standard_date = models.CharField(
+        "CE date override",
+        help_text="Manual override for automatically generated date or date range. "
+        + DocumentDateMixin.standard_date_helptext,
+        blank=True,  # use automatic date range from associated documents if this is blank
+        null=True,
+        max_length=255,
+        validators=[RegexValidator(DocumentDateMixin.re_date_format)],
+    )
+
+    def __str__(self):
+        """Use the name field for string representations"""
+        return self.name
+
+    @property
+    def date_str(self):
+        """Return a formatted string for the event's date/range, for use in public site"""
```

Comment:
it would be helpful to add a little more content to the docstring to make explicit the logic / fallback order here